premerge/advisor/advisor_lib.py

```diff
@@ -0,0 +1,25 @@
+from typing import TypedDict
+
+
+class TestFailure(TypedDict):
+    name: str
+    message: str
+
+
+class FailureExplanation(TypedDict):
+    name: str
+    explained: bool
+    reason: str | None
+
+
+def upload_failures(test_failures: list[TestFailure]):
```

Comment:
Won't you need to distinguish between failures from PRs and failures from the postsubmit buildbot? Won't you want a function parameter for that?

Reply:
Eventually, yes. This is just intended to be a skeleton so I can get the infrastructure up without needing a fully fleshed out implementation of the advisor.

Adding it now or later shouldn't make a big difference, so I can do whatever you prefer.

Reply:
Ok as is for now then.